src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private int size = 0;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
+    public ArrayList(int size) {
+        if (size <= DEFAULT_ARRAY_SIZE) {
+            array = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+        } else {
+            array = (T[]) new Object[size];
+        }
```

Comment:
You should have possibility to create ArrayList with capacity which is less than DEFAULT_ARRAY_SIZE